File: resQfood-user_management/src/main/java/models/ali/Line.java
package models.ali;

import java.sql.Date;

public class Line {
    // Attributes
    private int line_id;
    private int line_quantity;
    private int basket_id;
    private int product_id;
    private int user_id;
    private java.sql.Date line_date;

    private String name;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    // Constructors
    public Line(int line_id, int line_quantity, int basket_id, int product_id, int user_id, Date line_date) {
        this.line_id = line_id;
        this.line_quantity = line_quantity;
        this.basket_id = basket_id;
        this.product_id = product_id;
        this.user_id = user_id;
        this.line_date = line_date;
    }

    public Line() {}

    // Getter methods
    public int getLineId() {
        return line_id;
    }

    public int getLineQuantity() {
        return line_quantity;
    }

    public int getBasketId() {
        return basket_id;
    }

    public int getProductId() {
        return product_id;
    }

    public int getUserId() {
        return user_id;
    }

    public Date getLineDate() {
        return line_date;
    }

    public void setLineId(int line_id) {
        this.line_id = line_id;
    }

    // Setter methods (if needed)
    public void setLineQuantity(int line_quantity) {
        this.line_quantity = line_quantity;
    }

    // Additional methods (if needed)

    @Override
    public String toString() {
        return "Line [line_id=" + line_id + ", line_quantity=" + line_quantity +
                ", basket_id=" + basket_id + ", product_id=" + product_id +
                ", user_id=" + user_id + ", line_date=" + line_date + ", name=" + name + "]";
    }
}

File: src/Entity/Line.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

/**
 * Line
 *
 * @ORM\Table(name="line", indexes={@ORM\Index(name="fk1", columns={"basket_id"}), @ORM\Index(name="fk2", columns={"user_id"}), @ORM\Index(name="product_id", columns={"product_id"})})
 * @ORM\Entity
 */
class Line
{
    /**
     * @var int
     *
     * @ORM\Column(name="line_id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $lineId;

    /**
     * @var int
     *
     * @ORM\Column(name="line_quantity", type="integer", nullable=false)
     */
    private $lineQuantity;

    /**
     * @var int
     *
     * @ORM\Column(name="product_id", type="integer", nullable=false)
     */
    private $productId;

    /**
     * @var int
     *
     * @ORM\Column(name="basket_id", type="integer", nullable=false)
     */
    private $basketId;

    /**
     * @var int
     *
     * @ORM\Column(name="user_id", type="integer", nullable=false)
     */
    private $userId;

    /**
     * @var \DateTime|null
     *
     * @ORM\Column(name="line_date", type="date", nullable=true, options={"default"="CURRENT_TIMESTAMP"})
     */
    private $lineDate = 'CURRENT_TIMESTAMP';

    public function getLineId(): ?int
    {
        return $this->lineId;
    }

    public function getLineQuantity(): ?int
    {
        return $this->lineQuantity;
    }

    public function setLineQuantity(int $lineQuantity): static
    {
        $this->lineQuantity = $lineQuantity;

        return $this;
    }

    public function getProductId(): ?int
    {
        return $this->productId;
    }

    public function setProductId(int $productId): static
    {
        $this->productId = $productId;

        return $this;
    }

    public function getBasketId(): ?int
    {
        return $this->basketId;
    }

    public function setBasketId(int $basketId): static
    {
        $this->basketId = $basketId;

        return $this;
    }

    public function getUserId(): ?int
    {
        return $this->userId;
    }

    public function setUserId(int $userId): static
    {
        $this->userId = $userId;

        return $this;
    }

    public function getLineDate(): ?\DateTimeInterface
    {
        return $this->lineDate;
    }

    public function setLineDate(?\DateTimeInterface $lineDate): static
    {
        $this->lineDate = $lineDate;

        return $this;
    }


}
